client/components/domains/registrant-extra-info/fr-validate-contact-details.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { get, reduce, set } from 'lodash';
+import validatorFactory from 'is-my-json-valid';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import validationSchema from './fr-schema';
+
+const validate = validatorFactory( validationSchema, { greedy: true } );
+const debug = debugFactory( 'calypso:components:domains:registrant-extra-info:validation' );
+
+/*
+ * @returns errors by field, like: { 'extra.field: name, errors: [ string ] }
+ */
+export default function validateContactDetails( contactDetails ) {
+	// Populate validate.errors
+	validate( contactDetails );
+	validate.errors && debug( validate.errors );
+
+	return reduce(
+		validate.errors,
+		( acc, { field, message } ) => {
+			// Drop 'data.' prefix
+			const path = String( field ).split( '.' ).slice( 1 );
+
+			return set(
```

Comment:
Very hard to read, although smart. :laughing:

Reply:
Sorry, thanks :) I had another go, but feel free to object again - I'm familiar with the experience of coming back to code I wrote a month(/week/day) ago and scratching my head ;)